src/Pages/Music/Albums.elm

```diff
@@ -90,10 +167,86 @@ view model =
     { title = "Music.Artists"
     , body =
         [ row [ Element.height fill, Element.width fill ]
-            [ Components.VerticalNavMusic.view model.route
+            [ column [ Element.height fill, Element.width fill ]
+                [ Components.VerticalNavMusic.view model.route
+                , column [ Element.height fill, Element.width fill, paddingXY 20 30, Background.color Colors.sidebar, spacingXY 0 15 ]
+                    [ Element.text "SORT"
+                    , sortButton model.currentButton (Title Asc) "Title " TitleButtonMsg
+                    , sortButton model.currentButton (Year Asc) "Year " YearButtonMsg
+                    , sortButton model.currentButton (DateAdded Asc) "Date Added " DateButtonMsg
+                    , sortButton model.currentButton (Artist Asc) "Artist " ArtistButtonMsg
+                    , sortButton model.currentButton (Random Asc) "Random " RandomButtonMsg
+                    ]
+                ]
             , column [ Element.height fill, Element.width (fillPortion 6), paddingXY 0 0, spacingXY 5 7, Background.color Colors.background ]
                 [ Components.SectionHeader.viewAlbums model.album_list
                 ]
             ]
         ]
     }
+
+
+sortButton : AlbumSort -> AlbumSort -> String -> msg -> Element msg
+sortButton currentButton button name buttonMsg =
+    Input.button [ paddingXY 10 0 ]
+        { onPress = Just buttonMsg
+        , label =
+            case ( currentButton, button ) of
+                ( Title _, Title _ ) ->
```

Comment:
I think you should be able to get slightly nicer code, if you match for `Asc`|`Desc` here instead

Reply:
Basically, you could also have a function that handles the `row [ Font.color Colors.navTextHover ][ Element.text name ++ sortingOrder |> sortingOrderToString]` part.

You also might not have to be specific in both currentButton and button, you might be even able to ignore one of them?

Reply:
```
( Title sortingOrder, _ ) ->
	filterComponent name sortingOrder
```

Reply:
Hi @razzeee, 
I think that we need both of the parameters here `( Title _ , Title _ )  ->` , (**parameter1** _ , **parameter2** _), in order to highlight the currently selected button and toggle the signs when it’s clicked repeatedly as the `sortButton`, is a generalized function for creating sort buttons.

The idea behind passing the **model.currentButton** and **button**  to the `sortButton` function is to get the **currentButton** value and see which `sortButton` function call has that `button` (Title,DateAdded,etc). 

**For ex:** **`sortButton model.currentButton (Title Asc) "Title " TitleButtonMsg`**, it will highlight it’s color only when clicked and if we don’t do that then if one button is clicked all others will get highlighted, if I am not wrong.
 
And I also think that for handling `row [ Font.color Colors.navTextHover ][ Element.text name ++` part, using an if statement is a simpler way which will keep the code short and clean. Please let me know your thoughts on this.

Reply:
Not exactly happy with that either, but I think it's a bit more readable.

```

sortButtons : AlbumSort -> AlbumSort -> String -> msg -> Element msg
sortButtons currentButton button name buttonMsg =
    let
        isCurrentButton =
            case ( currentButton, button ) of
                ( Title _, Title _ ) ->
                    True

                ( DateAdded _, DateAdded _ ) ->
                    True

                ( Year _, Year _ ) ->
                    True

                ( Artist _, Artist _ ) ->
                    True

                ( Random _, Random _ ) ->
                    True

                _ ->
                    False
    in
    Input.button [ paddingXY 10 0 ]
        { onPress = Just buttonMsg
        , label =
            if currentButton == button then
                row [ currentButtonColor isCurrentButton ] [ Element.text (name ++ " ↓") ]

            else
                row [ currentButtonColor isCurrentButton ] [ Element.text (name ++ " ↑") ]
        }


currentButtonColor isCurrentButton =
    if isCurrentButton then
        Font.color Colors.navTextHover

    else
        Font.color Colors.navText
        
 ```

It also seems like it's not sorting correctly on first load?